Automattic/wp-calypso
# Signup: Trying siteless checkout

#### Changes proposed in this Pull Request

This PR explores siteless checkouts as a way of avoiding duplicate sites during signup. 

Current code is only a POC that modifies the default onboarding process so sites are not created when completing the last step (select a plan). 

At this point, the only remaining work is to make sure the transactions API endpoint creates the site once the payment has been processed.

Any assistance here would be highly appreciated.

Further reading: pbAPfg-5U-p2
Part of #39424

#### Testing instructions

* Apply D39598-code and sandbox the API.
* Go to `/start`.
* Complete all the steps and select a paid plan.
* Make sure you're redirected to the checkout page and the site has not been created.
* Make sure the checkout displays a shopping cart that contains the selected pain plan.
* **Do not proceed with the payment**. D39598-code still doesn't create sites after the plan is purchased, so you might end up adding the plan to a wrong site.
* Make sure the signup flow remains unmodified (sites created before checkout) when turning off the `signup/siteless-checkout` feature flag.
